### PR TITLE
[DR-3348] Remove search API endpoints

### PR DESCRIPTION
**Pull request by samanehsan**
Ticket: https://broadworkbench.atlassian.net/browse/DR-3348

The `elasticsearch-rest-high-level-client` library is out of date, but since it is no longer being used this PR removes the library and all of the code that depends on it (the search API controller endpoints).

There are several changes involved in removing elasticsearch:
1. Disable elasticsearch in dev (the only environment where this is enabled). This needs to get merged before any other changes:
- https://github.com/broadinstitute/terra-helmfile/pull/4906
- https://github.com/broadinstitute/datarepo-helm-definitions/pull/475
2. Remove the search API endpoints and elasticsearch library (this PR)
3. Remove the de-elasticsearch module from the datarepo helm chart: 
    https://github.com/broadinstitute/datarepo-helm/pull/247

### CODE REVIEW

File: src/main/java/bio/terra/app/configuration/ElasticSearchConfiguration.java [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-
-@Configuration
-public class ElasticSearchConfiguration {
-
-  @Bean
-  public RestHighLevelClient client(
-      @Value("${elasticsearch.hostname}") String hostname,
-      @Value("${elasticsearch.port}") int port) {
```
Comment (pshapiro4broad) [inline src/main/java/bio/terra/app/configuration/ElasticSearchConfiguration.java:17]: With these removed you should also remove the corresponding lines in `application.properties`:
```
features.search.api=disabled
elasticsearch.hostname=localhost
elasticsearch.port=9200
elasticsearch.numShards=3
```

**Review by pshapiro4broad — APPROVED** [review]
Looks good. Are there any extra steps needed to delete the ES instance in the cloud?

Also I think you can remove `ops/search/1000Genomes/upsert-metadata.py`

File: src/main/java/bio/terra/app/configuration/ElasticSearchConfiguration.java (hunk 1 above)
Comment (okotsopoulos) [reply src/main/java/bio/terra/app/configuration/ElasticSearchConfiguration.java:17]: Good shout, plus I wonder if these properties were overridden per environment too?  That could be another opportunity for clean-up.

**Review by okotsopoulos — APPROVED** [review]
The least vulnerable dependency is the one we remove :-) thanks!

Ditto Phil on wanting to know what other cleanup is required here.  I see some other PRs related to this, would you mind adding them to the description?

File: src/main/java/bio/terra/app/configuration/ElasticSearchConfiguration.java (hunk 1 above)
Comment (samanehsan) [reply src/main/java/bio/terra/app/configuration/ElasticSearchConfiguration.java:17]: Thanks for catching that!

**Comment by samanehsan** [comment]
> Are there any extra steps needed to delete the ES instance in the cloud?

@pshapiro4broad, this PR will disable the de-elasticsearch sub-chart for dev: https://github.com/broadinstitute/terra-helmfile/pull/4906

I think this will delete the elasticsearch deployment/resources created via helm but we can confirm once that is merged 👀

**Comment by samanehsan** [comment]
> Good shout, plus I wonder if these properties were overridden per environment too? That could be another opportunity for clean-up.

@okotsopoulos great point! There was a variable override for dev that I had missed: https://github.com/broadinstitute/terra-helmfile/pull/4906/files

**Pull request merged**